+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+
+                            x_ij_end = x_ij_end + opts.D_rk(kk+1)*x_ijk;
+                            
+                            if opts.g_path_at_stg
+                                obj.g.path(ii,jj,kk) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                            end
+                            if size(model.G_path, 1) > 0
+                                G_path = dcs.G_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                H_path = dcs.H_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                obj.G.path(ii,jj,kk) = {G_path};
+                                obj.H.path(ii,jj,kk) = {H_path};
+                            end
+                            if opts.cost_integration
+                                % also integrate the objective
+                                obj.f = obj.f + opts.B_rk(kk+1)*h*qj;
+                            end
+                        end
+                        if ~opts.right_boundary_point_explicit
+                            x_ijk = obj.w.x(ii,jj,opts.n_s+1);
+                            z_ijk = obj.w.z(ii,jj,opts.n_s+1);
+                            lambda_ijk = obj.w.lambda(ii,jj,opts.n_s+1);
+                            mu_not_c_ij = obj.w.mu_not_c(ii,jj);
+                            not_c_primals_ij = obj.w.not_c_primals(ii,jj);
+                            f_ijk = dcs.f_fun(x_ijk, z_ijk, ui, v_global, p);
+                            nabla_c_ijk = dcs.nabla_c_fun(x_ijk, z_ijk, v_global, p);
+                            lambda_cf_ijk = inv(nabla_c_ijk'*nabla_c_ijk)*nabla_c_ijk'*f_ijk;
+
+                            c = not_c_primals_ij(1); d = not_c_primals_ij(2);
+                            c_ijk = dcs.c_fun(x_ijk, z_ijk, v_global, p);
+                            obj.g.not_c(ii,jj) = {[c + d*c_ijk - 1;1-mu_not_c_ij(1) - mu_not_c_ij(2);mu_not_c_ij(1)*c_ijk + mu_not_c_ij(3)], [0;0;0], [inf;0;0]};
+                            obj.G.not_c(ii,jj) = {[c + d*c_ijk - 1;c;d]};
+                            obj.H.not_c(ii,jj) = {[mu_not_c_ij]};
+                            obj.g.lambda_explicit(ii,jj) = {[lambda_ijk-c*lambda_cf_ijk]};
+                            obj.g.dynamics(ii,jj,opts.n_s+1) = {x_ijk - x_ij_end};
+                            obj.g.c_lb(ii,jj,opts.n_s+1) = {dcs.c_fun(x_ijk, z_ijk, v_global, p), 0, inf};
+                            obj.g.z(ii,jj,opts.n_s+1) = {dcs.g_z_fun(x_ijk, z_ijk, ui, v_global, p)};
+                        end
+                        if ~opts.g_path_at_stg && opts.g_path_at_fe
+                            obj.g.path(ii,jj) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                        end
+                      case RKRepresentation.differential
+                        error("Differential representation without lifting is unsupported for gradient complementarity systems")
```

Comment:
instead of an error i would throw a warning and do differential_lift_x

Reply:
this also does not allow Lobbato methods. 
however, is it possible to have them in lifted_x form?

Reply:
In prior implementation this was explicitly disallowed. I spent some time looking into it and I am under the impression that is is not possible due to their structure, but I am not 100% on this.